TFNetworkLayer.py

```diff
@@ -882,6 +892,46 @@ def batch_norm(self, data,
         bn += beta
       return bn
 
+  def fused_batch_norm(self, data,
+                       use_shift=True, use_std=True,
+                       momentum=0.99, epsilon=1e-3,
+                       gamma_init="ones"):
+    """
+    :param Data data:
+    :param bool use_shift:
+    :param bool use_std:
+    :param float momentum: for the running average of sample_mean and sample_std
+    :param float epsilon:
+    :param str gamma_init: gamma initializer
+    :rtype: tf.Tensor
+
+    Also see:
+      tf.layers.batch_normalization
+      https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization
+    """
+    x = data.get_placeholder_as_batch_major()
+    axis = data.feature_dim_axis
+    training = self.network.train_flag
+    with self.var_creation_scope():
+      if gamma_init == "zeros":
+        gamma_initializer = tf.zeros_initializer()
+      else:
+        gamma_initializer = tf.ones_initializer()
+      bn = tf.layers.batch_normalization(
+        inputs=x, axis=axis, gamma_initializer=gamma_initializer,
+        momentum=momentum, epsilon=epsilon, center=use_shift,
+        scale=use_std, training=training, fused=True)
+
+    import re
+    scope_name_prefix = tf.get_variable_scope().name + "/"  # e.g. "layer1/"
+    params = tf.get_collection(tf.GraphKeys.GLOBAL_VARIABLES, scope=re.escape(scope_name_prefix))
+    for p in params:
+      if not p.name.startswith(scope_name_prefix):
+        continue
+      assert p.name.startswith(scope_name_prefix) and p.name.endswith(":0")
+      self.params[p.name[len(scope_name_prefix):-2]] = p
+    return bn
+
```

Comment:
Remove the function `LayerBase.fused_batch_norm`.
Directly use `tf.nn.fused_batch_norm` in the `LayerBase. batch_norm` function. Also the complex parameter handling is not needed when you directly use `tf.nn.fused_batch_norm`.

Reply:
Ok